### PR TITLE
Misc transformer fixes - 2

### PR DESCRIPTION
**Pull request by hariharans29**
### Description
1. The graph pattern search introduced in https://github.com/microsoft/onnxruntime/pull/13914/ needs to be enhanced so that SkipLayerNormalization is supported

2. Fix fp32 parity for GPT-2 while using `SkipLayerNormalization` fusion. The optional output of SLN needs to also include the bias (if present) and the added output should be a sum of `input + skip + (bias)`

### Motivation and Context
Fix some breaking tests

### CODE REVIEW

File: onnxruntime/core/graph/contrib_ops/bert_defs.cc [hunk 1]
```diff
@@ -389,7 +389,7 @@ ONNX_MS_OPERATOR_SET_SCHEMA(SkipLayerNormalization, 1,
                                 .Output(0, "output", "3D output tensor with shape (batch_size, sequence_length, hidden_size)", "T")
                                 .Output(1, "mean", "Saved mean used during training to speed up gradient computation", "U", OpSchema::Optional)
                                 .Output(2, "inv_std_var", "Saved inverse standard variance used during training to speed up gradient computation.", "U", OpSchema::Optional)
-                                .Output(3, "input_skip_sum", "Sum of the input and skip inputs with shape (batch_size, sequence_length, hidden_size).", "T", OpSchema::Optional)
+                                .Output(3, "input_skip_bias_sum", "Sum of the input and skip inputs (and bias if it exists) with shape (batch_size, sequence_length, hidden_size).", "T", OpSchema::Optional)
```
Comment (hariharans29) [inline onnxruntime/core/graph/contrib_ops/bert_defs.cc:392]: Will update docs

File: onnxruntime/python/tools/transformers/convert_generation.py [hunk 2]
```diff
@@ -1134,54 +1136,133 @@ def generate_gpt2_init_decoder(
         [0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0],
     )
 
+    # Normalization Node is : SkipLayerNormalization
+    if logits_matmul_to_residual_add_path is None:
```
Comment (hariharans29) [inline onnxruntime/python/tools/transformers/convert_generation.py:1140]: Will think of a way to add a test for this in the next round of misc changes (have some more in the backlog). This is just to quickly unblock main branch.

File: onnxruntime/python/tools/transformers/models/gpt2/gpt2_helper.py [hunk 3]
```diff
@@ -522,6 +522,11 @@ def optimize_onnx(
             # Potentially, we can add other optimization like unpad for effective transformer
             optimization_options.attention_mask_format = AttentionMaskFormat.MaskIndexEnd
 
+        # TODO(hasesh): Investigate parity issue for GPT-2 fp16 when SkipLayerNormalization
+        # is enabled
+        if is_float16:
+            optimization_options.enable_skip_layer_norm = False
```
Comment (hariharans29) [inline onnxruntime/python/tools/transformers/models/gpt2/gpt2_helper.py:528]: It seems like there are some parity issues with fp16 + SLN. I added the SLN op to the op_block_list like this - arguments.extend(["--op_block_list", "Add", "LayerNormalization", "SkipLayerNormalization", "FastGelu"]) which should in theory still make it the same as the graph with LayerNormalizations + Adds. Am I missing something obvious: @tlwu @wangyems ?

Comment (tianleiwu) [reply onnxruntime/python/tools/transformers/models/gpt2/gpt2_helper.py:528]: The order of Add might matter. In the original onnx graph, the order of add is input + bias + skip, in your SLN kernel it is like input + skip + bias. When fp16 is used, the result might be different.

**Review by yufenglee — APPROVED** [review]

**Comment by yufenglee** [comment]
Let's merge this PR firstly while continuing investigating the fp16 parity issue on some tests.

**Pull request merged**